planet/src/main/webapp/js/authentication.js

```diff
@@ -22,18 +42,35 @@ function resetForm() {
     }
 }
 
-function submitForm() {
+function signUp() {
     const password = document.getElementById('password').value;
     const passwordConfirmation = document.getElementById('repeat-password').value;
     const email = document.getElementById('email').value;
+    const displayName = document.getElementById('first_name').value 
+    + ' ' + document.getElementById('last_name').value;
+    resetForm('input-sign-up');
     if (password !== passwordConfirmation) {
-        resetForm();
         return;
     }
-    firebase.auth().createUserWithEmailAndPassword(email, password).catch(function(error) {
+    firebase.auth().createUserWithEmailAndPassword(email, password)
+    .then(function() {
+        return firebase.auth().currentUser.updateProfile({
+            displayName: displayName,
+        })
+    }).catch(function(error) {
+        // Handle Errors here.
+        console.log(error);
+        // ...
```

Comment:
For errors that require the user to take action, we should tell the user about them. An alert is probably good enough for the prototype.

https://firebase.google.com/docs/reference/js/firebase.auth.Auth#createuserwithemailandpassword
https://firebase.google.com/docs/reference/js/firebase.auth.Auth#signinwithemailandpassword

Reply:
Okay! I'll add that right away